src/Blocks/BlockInfested.h

```diff
@@ -0,0 +1,73 @@
+
+// BlockInfested.h
+
+/* This Block Handler describes the blocks spawning silver fishes. Mojang calls them monster egg */
+
+class cBlockInfestedHandler:
+	public cBlockHandler
+{
+	using Super = cBlockHandler;
+
+public:
+
+	using Super::Super;
+
+private:
+
+	virtual cItems ConvertToPickups(NIBBLETYPE a_BlockMeta, const cEntity * a_Digger, const cItem * a_Tool) const override
+	{
+		switch (a_BlockMeta)
+		{
+			case E_META_SILVERFISH_EGG_STONE:
+			{
+				if (a_Tool->m_Enchantments.GetLevel(cEnchantments::enchSilkTouch))
+				{
+					return cItem(E_BLOCK_STONE);
+				}
+				else
+				{
+					return cItem(E_BLOCK_COBBLESTONE);
+				}
+			}
+			case E_META_SILVERFISH_EGG_COBBLESTONE:          return cItem(E_BLOCK_COBBLESTONE);
+			case E_META_SILVERFISH_EGG_STONE_BRICK:          return cItem(E_BLOCK_STONE_BRICKS);
+			case E_META_SILVERFISH_EGG_MOSSY_STONE_BRICK:    return cItem(E_BLOCK_STONE_BRICKS, 1, E_META_STONE_BRICK_MOSSY);
+			case E_META_SILVERFISH_EGG_CRACKED_STONE_BRICK:  return cItem(E_BLOCK_STONE_BRICKS, 1, E_META_STONE_BRICK_CRACKED);
+			case E_META_SILVERFISH_EGG_CHISELED_STONE_BRICK: return cItem(E_BLOCK_STONE_BRICKS, 1, E_META_STONE_BRICK_ORNAMENT);
+		}
+		return {};
+	}
+
+
+
+
+
+	virtual void OnPlayerBrokeBlock(
+			cChunkInterface & a_ChunkInterface, cWorldInterface & a_WorldInterface,
+			cPlayer & a_Player,
+			Vector3i a_BlockPos,
+			BLOCKTYPE a_OldBlockType, NIBBLETYPE a_OldBlockMeta
+	) const override
+	{
+		SpawnSilverfish(a_ChunkInterface, a_WorldInterface, a_BlockPos);
```

Comment:
This should only spawn for creative players.

Reply:
you are right.

Reply:
Is there function that is call that the player is not broken by a player but by an explosion?

Reply:
In explodinator.cpp but I think it uses setblock

Reply:
`OnBroken` I think.

Reply:
i'm not trying to break some blocks i want to know if the block was broken by an explosion :D

Reply:
hm but then we can't prevent spawning them, when the player is in creative

Reply:
I tested and creative mode doesn't make a difference for explosions, the silverfish were always spawned.

Reply:
So the Blocks break under three conditions:

- being mined by a player in survival
- being destroyed by an explosion
- a nearby silverfish is attacked.


As far as I understood OnBreak is triggered before OnPlayerBreak and if a player breaks the block both get triggered.

So if we do the replacement in OnBreak and OnPlayerBreak we spawn two entities.

If we do the check only in OnBreak we can't prevent them from spawning if the player is in creative mode. If we the check only in OnBreak we can't spawn them as a result from an explosion.

Reply:
I didn't realise both were called when a player breaks the block. That's not very helpful. 
It might require some reworking of the block handler calls to make this work.

Reply:
Yeah - i checked when the on break function is called... it's a bit of a mess